components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.email.mgt.store;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.email.mgt.internal.I18nMgtDataHolder;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerServerException;
+import org.wso2.carbon.identity.governance.model.NotificationTemplate;
+import org.wso2.carbon.identity.governance.service.notification.NotificationChannels;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for storing notification templates from the email-admin-config.xml file.
+ * This class is primarily used to provide default notification templates without addition/modification,
+ * allowing for quick access of templates without migration.
+ * Templates are stored in nested maps, where the outer map is keyed by the template display name and the inner map
+ * is keyed by locale. This allows for easy retrieval of localized templates.
+ * This class used by {@link HybridTemplateManager} as a fallback mechanism.
+ * This class only supports for tenant specific notification templates.
```

Comment:
`This class only supports for tenant specific notification templates.`
This comment is incorrect right?

Reply:
It is correct, since we do not have an in-memory concern for application specific templates. They always can be retrieved from the DB

Reply:
as @JeethJJ said with current logic we are not supporting application specific templates. But with some alterations we can support application specific templates. Since this application feature is introduced recently, we can improve the logic to send default template when we send GET request for application specific tempate.
@darshanasbg WDYT?

Reply:
Aa ok.. Im not really referring to the application specific templates.

What I had in mind was `InMemoryBasedTemplateManager` not maintaining even tenant **specific** templates.. `InMemoryBasedTemplateManager` is keeping set of templates that will be commonly used as default templates across all the tenants.

Reply:
I revisited the existing code 
ref: https://github.com/wso2-extensions/identity-event-handler-notification/blob/master/components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java#L724

Here when there is no application template found, it fall backs to organisation wise template, to support this behaviour we have to keep the logic as it is.  Other wise, customer has to configure email template for each and every application. Hence +1 to have this logic to skip when application id presented.

Reply:
Ack

Reply:
@Thumimku Im not exactly clear on the change you are proposing.

Note that my intention is only to correct the quoted the class comment to be more accurate, not to do any logic change of the code execution.

Reply:
Updated the comment.

Reply:
To be more direct, the class comment can be reworded as,

/**
 * This class is responsible for handling default templates provided by the system.
 * 
 * System default templates are used as default org notification templates in a given tenant and based on the content of the email-admin-config.xml.
 * Any modifications to system default templates needs to be done via email-admin-config.xml and any such changes will be reflected only with the server startup.
 * 
 * This class have support for get, list, exists operations and any invocations to modification operations throws NotImplementedException.
 * 
 * This class expected to be use only with conjunction with another {@link TemplatePersistenceManager} implementation which supports full CRUD operations, hence {@link DefaultTemplateManager} provides that aggregation using this as a fallback provider.
 */

Reply:
Updated with https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/commits/5d985d4254d5681ad87d157dc8a8a8c37cd69ff9